### ISSUE TITLE
Part of the pagination doesn't work

### ISSUE THREAD

**Issue by lio1972** [daniel-nagy/md-data-table#469]
The part of my pagination that is showing 
1 - 5 of 9
doesn't work. What I am doing wrong?

`myApp.service('allCurrentSettingsService', ['$http', '$q', function ($http, $q) {
            var allSettings = null;
            this.getList = function () {
                var def = $q.defer()
                if (allSettings) {
                    def.resolve(allSettings);
                } else {
                    $http.post('GetAllCurrentSettings')
                      .then(function (response) {
                          var response = $.parseJSON(response.data)
                          allSettings = response;
                          def.resolve(allSettings);
                      });
                }
                return def.promise;
            }
        }]);`

> ` <md-table-pagination md-limit="query.limit" md-limit-options="limitOptions" md-page="query.page" md-total="{{allSettings.length}}" md-page-select="options.pageSelect" md-boundary-links="options.boundaryLinks" md-on-paginate="logPagination"></md-table-pagination>
       `

**Comment by daniel-nagy** [comment]
I don't see `query` defined on the scope.
I don't see `allSettings` on the scope.

You'll need to do something like the following.

```javascript
myApp.service('allCurrentSettingsService', ['$http', '$q', '$scope', function ($http, $q, $scope) {
  var allSettings = null;

  $scope.total = 0;
  $scope.query = {
    limit: 10,
    page: 0
  };

  this.getList = function () {
    var def = $q.defer();

    if(allSettings) {
      return def.resolve(allSettings);
    }

    $http.post('GetAllCurrentSettings', $scope.query).then(function (response) {
      allSettings = angular.fromJson(response.data);
      def.resolve(allSettings);
    });
    
    return def.promise;
  }

  this.getList().then(function (settings) {
    $scope.total = settings.length;
  });
}]);
```

Saving the result and returning form `getList` doesn't really makes sense if you are trying to do pagination.

**Comment by lio1972** [comment]
Hi and thank you for the help. I followed you you suggestion but I am getting the following error:
Unknown provider: $scopeProvider <- $scope <- allCurrentSettingsService


 myApp.service('allCurrentSettingsService', ['$http', '$q', '$scope', function ($http, $q, $scope)         {  var allSettings = null;             $scope.total = 0;            $scope.query = {    limit: 10,    page: 0  };            this.getList = function ()            {                var def = $q.defer();                 if(allSettings) {                     return def.resolve(allSettings);                }               $http.post('GetAllCurrentSettings', $scope.query).then(function (response)            {                  allSettings = angular.fromJson(response.data);                  def.resolve(allSettings);    });               return def.promise;  }              this.getList().then(function (settings)             {               $scope.total = settings.length;             });        }]);


myApp.controller('myController', ['$scope', 'companiesService', 'allCurrentSettingsService', 'deleteService', 'setNewRecordService', 'getDocTypesService', '$timeout',          function ($scope, companiesService, allCurrentSettingsService, deleteService, setNewRecordService, getDocTypesService, $timeout) {              $scope.limitOptions = [5, 10, 15];
              $scope.query = {                  order: 'name',                  limit: 10,                  page: 1              };
              $scope.toggleLimitOptions = function () {                  $scope.limitOptions = $scope.limitOptions ? undefined : [5, 10, 15];              };
              $scope.logPagination = function (page, limit) {                  console.log('page: ', page);                  console.log('limit: ', limit);              }
              $scope.logOrder = function (order) {                  console.log('order: ', order);              };                           $scope.selections = { company: null, document: null };
              $scope.currentSettings = '';              companiesService.getList().then(function (value) {                  $scope.currentSettings = value;                                });
              $scope.allSettings = '';              allCurrentSettingsService.getList().then(function (value) {                  $scope.allSettings = value;              });
              $scope.deleteRecordFromDB = function (recId, compName, custName, docName) {                  deleteService.removeRow(recId, compName, custName, docName);              };
              $scope.companyName = '';              $scope.setNewRecord = function () {                  setNewRecordService.setNewRecord($scope.selected.company, $scope.enteredCustomer, $scope.selected.document)              };
              $scope.getTypes = function () {                  getDocTypesService.getDocTypes($scope.selected.company, $scope.enteredCustomer).then(function (value) {                      $scope.docSettings = value                  });              };          }        ]); 

   

 On Thursday, September 1, 2016 3:23 PM, Daniel Nagy <notifications@github.com> wrote:
 

 Closed #469.—

**Comment by daniel-nagy** [comment]
Oh I'm sorry, I didn't realize it was a service. You can't inject `$scope` Into a service. You'll need to manage the scope within your controller.
